### ISSUE TITLE
Infinite recursion in SQLite grammar

### ISSUE THREAD

**Issue by Changochen** [antlr/grammars-v4#2266]
As shown in https://github.com/antlr/grammars-v4/blob/cda43d1e0e58baccdf62da56d01b86ab76e88d93/sql/sqlite/SQLiteParser.g4#L605,  the rule seems wrong as it will never stop expanding.

```antlr
order_by_expr_asc_desc:
    ORDER_ BY_ order_by_expr_asc_desc
;
```

**Comment by Changochen** [comment]
I think it should be
```antlr
order_by_expr_asc_desc:
    ORDER_ BY_ expr_asc_desc
;
```

**Comment by kaby76** [comment]
Dup of https://github.com/antlr/grammars-v4/issues/2039

The main problem is that it's a useless symbol. It is used only by `window_function`, which is used only in `window_function_invocation`---and that symbol is not used anywhere. Trash's tranalyze has graph analysis of symbol uses and finds all useless symbols. I think Bison has a similar feature.

In fact, there's an ominous "// unknown" comment right before this production.

**Comment by dpmm99** [comment]
The original issue was fixed [here](https://github.com/antlr/grammars-v4/commit/76b1d66d768902f821b7a7440507bfdb9b6a32f9); do you want to rename this issue because of the unused rule or open a new issue?

**Comment by kaby76** [comment]
I think close this issue out. All grammars should be analyzed for useless rules, so the issue is not limited to this grammar alone.

**Comment by dpmm99** [comment]
Makes sense. Also, I just noticed #1493 covers the window functions not being supported.
